packages/taquito/src/estimate/rpc-estimate-provider.ts

```diff
@@ -457,4 +458,64 @@ export class RPCEstimateProvider extends Provider implements EstimationProvider
     }
     return Estimate.createEstimateInstanceFromProperties(estimateProperties);
   }
+
+  /**
+   *
+   * @description Estimate gasLimit, storageLimit and fees for stake operation
+   *
+   * @returns An estimation of gasLimit, storageLimit and fees for the stake operation
+   *
+   * @param Estimate
+   */
+  async stake(params: StakingParams) {
+    const protocolConstants = await this.context.readProvider.getProtocolConstants('head');
```

Comment:
Same as `rpc-contract-provider.ts` we will want to validate amount. Estimate Provider can be use solely we need to validate what's being passed separately